Formula/gammalib.rb

```diff
@@ -0,0 +1,41 @@
+class Gammalib < Formula
+  desc "Toolbox for the analysis of astronomical gamma-ray data"
+  homepage "http://cta.irap.omp.eu/gammalib/about.html"
+  url "http://cta.irap.omp.eu/ctools/releases/gammalib/gammalib-1.6.3.tar.gz"
+  sha256 "b93bc03562ba1fcb21ab6f48943f9c28230c04d1b0ef986569c88b05d8e9dbbe"
+  revision 1
+
+  depends_on "cfitsio"
+  depends_on "doxygen"
+  depends_on "ncurses"
+  depends_on "python"
+  depends_on "readline"
+  depends_on "swig"
+
+  def install
+    system "./configure", "PYTHON=/usr/local/bin/python3", "--prefix=#{prefix}"
+    system "make"
+    system "make", "install"
+  end
+
+  def caveats; <<~EOS
+    Because gammalib depends on several installation-dependent
+    environment variables to function properly, you should
+    add the following commands to your shell initialization
+    script or call them directly
+    before using gammalib.
+
+    For zsh users:
+      export GAMMALIB=/usr/local/
+      source $GAMMALIB/bin/gammalib-init.sh
+  EOS
+  end
+
+  test do
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.